### PR TITLE
Add concurrent option for executing tasks

### LINKED ISSUE #18: Parallel Tasks
Great lib! I'm running into a few cases where I would like to execute tasks in parallel. Would be great if there was support for this. One idea is options object to the `Listr` constructor function such as `{ parallel: true }`. I'm working right now to see if I can hack this together or drop down to `ora` and do it just with that lib. Will report back here if I find a good solution.

### PR DESCRIPTION
**Pull request by southpolesteve**
Fixes #18

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -50,7 +51,13 @@ class Listr {
 	run() {
 		this.render();
 
-		return this._tasks.reduce((promise, task) => promise.then(() => task.run()), Promise.resolve())
+		const tasks = this._options.parallel === true ? (
+			Promise.all(this._tasks.map(task => task.run()))
+		) : (
+			this._tasks.reduce((promise, task) => promise.then(() => task.run()), Promise.resolve())
+		);
```
Comment (SamVerschueren) [inline index.js]: Use a normal if else statement instead of the ternary operator.

**Comment by southpolesteve** [comment]
Fixed and rebased

File: index.js [hunk 2]
```diff
@@ -17,7 +17,8 @@ class Listr {
 		this._RendererClass = CLIRenderer;
 		this._tasks = [];
 		this._options = Object.assign({
-			showSubtasks: true
+			showSubtasks: true,
+			parallel: false
```
Comment (sindresorhus) [inline index.js]: `parallel` => `concurrent`

http://stackoverflow.com/questions/1050222/concurrency-vs-parallelism-what-is-the-difference

Comment (SamVerschueren) [inline index.js]: :+1: otherwhise,  looks good to me

**Comment by southpolesteve** [comment]
changed and rebased

**Comment by SamVerschueren** [comment]
Super nice! Thanks a lot @southpolesteve for this PR :).

**Pull request closed without merging**